realm/realm-library/src/main/java/io/realm/internal/SharedGroup.java

```diff
@@ -293,6 +297,29 @@ public int hashCode() {
         }
     }
 
+    /**
+     * Waits for change committed by {@link SharedGroup} in other Thread.
+     *
+     * @return {@code true} if successfully detects change, {@code false} no change has detected otherwise.
+     */
+    public boolean waitForChange() {
+        if (isStoppedWaitingByUser) {
+            isStoppedWaitingByUser = false;
+            return false;
+        }
```

Comment:
@beeender 

There was [a discussion](https://github.com/realm/realm-wiki/wiki/Java-Multi-threaded-wait_for_change) on how to implement this PR. Please take a look & add comments there.